wd_helper.au3

```diff
@@ -85,13 +87,15 @@ Global Enum _ ; _WD_FrameList() , _WD_FrameListFindElement()
 		$_WD_FRAMELIST_MatchedElements = 6, _ ; array of matched element from _WD_FrameListFindElement()
 		$_WD_FRAMELIST__COUNTER
 
-Global Enum _ ; https://www.w3schools.com/jsref/prop_doc_readystate.asp
+#Tidy_ILC_Pos=42
+Global Enum _                            ; https://www.w3schools.com/jsref/prop_doc_readystate.asp
 		$_WD_READYSTATE_Uninitialized, _ ; Has not started loading
 		$_WD_READYSTATE_Loading, _       ; Is loading
 		$_WD_READYSTATE_Loaded, _        ; Has been loaded
 		$_WD_READYSTATE_Interactive, _   ; Has loaded enough to interact with
 		$_WD_READYSTATE_Complete, _      ; Fully loaded
 		$_WD_READYSTATE__COUNTER
+#Tidy_ILC_Pos=0
 
```

Comment:
When I run Tidy with these settings, the ILC aren't alined at pos 42. They are instead at pos 50, which appears to be due to the leading tabs on lines 92-97.

Reply:
I have correct position on 42 column
![image](https://github.com/Danp2/au3WebDriver/assets/11089482/04674048-1fff-49fe-a784-5628e544faee)

Reply:
please show me your TIDY.ini settings.

Here are mine:
```
[ProgramSettings]
 * * *  Update Functions/Keywords/Macros/UDFs to Proper case
proper=1

 * * * Update Constants defined in the "constant Include files"
properconstants=1

#-->Generate Documentation file for the Program and optionally Display it with Notepad at the end ot the Tidy run
Gen_Doc=0
Gen_Doc_Show=0


 * * *  Indent:  0 = Tabs          >0 = Number of Space            -1 = strip space between code and comment
tabchar=0
 * * *  #-->Define the default column for inline comments, 0=keep current position
icl_pos=0
 * * *  #-->Set default for Skip_CommentBlock  1=skip, 0=Tidy Comment blocks  default=1
Skip_commentblock=0

#--> Tidy commentblock 1=Tidy inside commentblock, 0=only indent whole block, -1=leave all alone .. default=0
Tidy_commentblock=-1

```

Reply:
I don't have a tidy.ini file. I temporarily created one using your settings, and then the indentation was working as expected. Strangely, I removed the tidy.ini file and now the indentation still works as expected. 🤔

Reply:
Scite restarted ?

Reply:
We can add all this settings via directive

Reply:
No, I didn't know that it would retain the settings between each Tidy run.  This is what I get when I use VSC to call Tidy --

![image](https://github.com/Danp2/au3WebDriver/assets/7101313/9ec5acae-d596-4501-a227-d98622faba41)

Strangely, it performs as expected when running in VSC *if* Scite is loaded in the background. I'll have to do some digging to see if Scite is creating a temporary copy of tidy.ini somewhere.

Reply:
Check tidy version for both setups.

Reply:
They run the exact same copy of Tidy. Here's the output from each one --

```text
>"C:\Program Files (x86)\AutoIt3\SciTE\..\AutoIt3.exe" "C:\Program Files (x86)\AutoIt3\SciTE\AutoIt3Wrapper\AutoIt3Wrapper.au3" /Tidy /in "C:\Users\danpo\Downloads\genericcdp.au3"
+>13:02:29 Starting AutoIt3Wrapper (23.402.1150.0) from:SciTE.exe (5.3.5.0)  Keyboard:00000409  OS:WIN_11/2009  CPU:X64 OS:X64  Environment(Language:0409)  CodePage:0  utf8.auto.check:4
+>         SciTEDir => C:\Program Files (x86)\AutoIt3\SciTE   UserDir => C:\Users\danpo\AppData\Local\AutoIt v3\SciTE\AutoIt3Wrapper   SCITE_USERHOME => C:\Users\danpo\AppData\Local\AutoIt v3\SciTE 
>Running Tidy (23.402.1150.0)  from:C:\Program Files (x86)\AutoIt3\SciTE\tidy  C:\Users\danpo\Downloads\genericcdp.au3
+>13:02:29 Tidy ended, no changes made.  rc:-1
+>13:02:29 AutoIt3Wrapper Finished.
>Exit code: 0    Time: 0.7501
```

```text
#1: Starting process #1
"C:\Program Files (x86)\AutoIt3\AutoIt3.exe" "C:\Program Files (x86)\AutoIt3\SciTE\AutoIt3Wrapper\AutoIt3Wrapper.au3" /Tidy /in "c:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\wd_helper.au3" [PID 5552]
    +>14:03:31 Starting AutoIt3Wrapper (23.402.1150.0) from:Code.exe (1.81.1.0)  Keyboard:00000409  OS:WIN_11/2009  CPU:X64 OS:X64  Environment(Language:0409)
    >Running Tidy (23.402.1150.0)  from:C:\Program Files (x86)\AutoIt3\SciTE\tidy  c:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\wd_helper.au3
    - Unable to restore Folds and Boomarks due to Sessionfile not found:C:\Users\danpo\AppData\Local\Temp\tidyscite.session
    +>14:03:32 Tidy ended. rc:0
    +>14:03:32 AutoIt3Wrapper Finished.
    >>Exit code 0 Time: 1.603
```

Ignore the "Unable to restore Folds and Boomarks" message as I don't believe it is relevant.

Reply:
Check
#Tidy_Parameters=/tcb=-1 /SCI=9

Reply:
### VSC

```
#1: Starting process #1
"C:\Program Files (x86)\AutoIt3\AutoIt3.exe" "C:\Program Files (x86)\AutoIt3\SciTE\AutoIt3Wrapper\AutoIt3Wrapper.au3" /Tidy /in "c:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\wd_helper.au3" [PID 17472]
    +>14:41:33 Starting AutoIt3Wrapper (23.402.1150.0) from:Code.exe (1.81.1.0)  Keyboard:00000409  OS:WIN_11/2009  CPU:X64 OS:X64  Environment(Language:0409)
    >Running Tidy (23.402.1150.0)  from:C:\Program Files (x86)\AutoIt3\SciTE\tidy  c:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\wd_helper.au3
    0.03 Creating Tables
    0.03 SciTE_File$:C:\Program Files (x86)\AutoIt3\SciTE\tidy\..\api\au3.api Exist: 1
    0.03 Tidy_File$:C:\Users\danpo\AppData\Local\AutoIt v3\SciTE\Tidy\au3.api Exist: 1
    0.03 functions_File$:C:\Users\danpo\AppData\Local\AutoIt v3\SciTE\Tidy\functions.tbl Exist: 1
    0.03 CompareFileTime(&SciTE_File_Date, &Tidy_File_Date): 0
    0.03 CompareFileTime(&functions_File_Date, &Tidy_File_Date): 0
    0.03 Files are same date, so no need to rebuild.
    0.03 Start Pre-processing File...
    0.06 Pre-processing record: 1000
    0.09 Pre-processing record: 2000
    0.11 Pre-processing record: 3000
    0.13 Start Processing File...
    0.25 Processing record: 1000
    0.39 Processing record: 2000
    0.53 Processing record: 3000
    0.61 Processing done ... show possible warnings.
    0.61 Start backup process. TargetDi$=c:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\BackUp\
    0.61 Find last backup file name. KeepNVersions%= 0
    0.61 Find last backup file name. seqnum= 5
    0.61 Read full processed file.
    0.64 Disable SciTE folds !
    0.64 Copy Oldfile to c:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\BackUp\wd_helper_old5.au3
    0.64 Write updated script to c:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\wd_helper.au3
    0.64 cleanup temp file C:\Users\danpo\AppData\Local\Temp\Tid1BC2.tmp
    0.64 Process KeepNVersions: 0
    - Unable to restore Folds and Boomarks due to Sessionfile not found:C:\Users\danpo\AppData\Local\Temp\tidyscite.session
    0.64 Done...
    +>14:41:34 Tidy ended. rc:0
    +>14:41:34 AutoIt3Wrapper Finished.
    >>Exit code 0 Time: 1.466
```

### Scite

```
>"C:\Program Files (x86)\AutoIt3\SciTE\..\AutoIt3.exe" "C:\Program Files (x86)\AutoIt3\SciTE\AutoIt3Wrapper\AutoIt3Wrapper.au3" /Tidy /in "C:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\wd_helper.au3"
+>14:46:35 Starting AutoIt3Wrapper (23.402.1150.0) from:SciTE.exe (5.3.5.0)  Keyboard:00000409  OS:WIN_11/2009  CPU:X64 OS:X64  Environment(Language:0409)  CodePage:0  utf8.auto.check:4
+>         SciTEDir => C:\Program Files (x86)\AutoIt3\SciTE   UserDir => C:\Users\danpo\AppData\Local\AutoIt v3\SciTE\AutoIt3Wrapper   SCITE_USERHOME => C:\Users\danpo\AppData\Local\AutoIt v3\SciTE 
>Running Tidy (23.402.1150.0)  from:C:\Program Files (x86)\AutoIt3\SciTE\tidy  C:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\wd_helper.au3
0.02 Creating Tables
0.02 SciTE_File$:C:\Program Files (x86)\AutoIt3\SciTE\tidy\..\api\au3.api Exist: 1
0.02 Tidy_File$:C:\Users\danpo\AppData\Local\AutoIt v3\SciTE\Tidy\au3.api Exist: 1
0.02 functions_File$:C:\Users\danpo\AppData\Local\AutoIt v3\SciTE\Tidy\functions.tbl Exist: 1
0.02 CompareFileTime(&SciTE_File_Date, &Tidy_File_Date): 0
0.02 CompareFileTime(&functions_File_Date, &Tidy_File_Date): 0
0.02 Files are same date, so no need to rebuild.
0.02 Start Pre-processing File...
0.03 Pre-processing record: 1000
0.05 Pre-processing record: 2000
0.06 Pre-processing record: 3000
0.06 Start Processing File...
0.16 Processing record: 1000
0.23 Processing record: 2000
0.31 Processing record: 3000
0.36 Processing done ... show possible warnings.
0.36 Start backup process. TargetDi$=C:\Users\danpo\OneDrive\Documents\GitHub\WebDriver\BackUp\
0.36 Find last backup file name. KeepNVersions%= 0
0.36 Find last backup file name. seqnum= 1
0.36 Read full processed file.
0.38 Tidy AutoIt3 finished and has no updates to make!
0.38 Done...
+>14:46:36 Tidy ended, no changes made.  rc:-1
+>14:46:36 AutoIt3Wrapper Finished.
>Exit code: 0    Time: 1.1
```

Reply:
Based on this line, Tidy is clearly making changes based on the environment --

`    0.64 Disable SciTE folds !`

Reply:
Looking at AutoIt3Wrapper.au3, it contains coding to detect if it was launched by Scite, so that may be contributing to the differences I'm experiencing.

Reply:
https://www.autoitscript.com/forum/topic/210716-using-tidy-with-other-editors/?do=findComment&comment=1523180

Reply:
Resolved with updated Tidy beta